src/test/java/org/terasology/launcher/game/TestRunGameTask.java

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.game;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import javafx.concurrent.WorkerStateEvent;
+import javafx.util.Pair;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.spf4j.log.Level;
+import org.spf4j.test.log.LogAssert;
+import org.spf4j.test.log.TestLoggers;
+import org.spf4j.test.matchers.LogMatchers;
+import org.terasology.launcher.SlowTest;
+import org.testfx.framework.junit5.ApplicationExtension;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.not;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+@Timeout(5)
+@ExtendWith(ApplicationExtension.class)
+public class TestRunGameTask {
```

Comment:
Also not reviewing this in detail. I still think that testing the game log bridging and introducing complex dependency mangling for that is a bit of overkill, but I'm impressed you pulled this off. As talked about earlier, as long as we can keep these tests without too much maintenance effort and investment I'm fine with having them, but I don't want them to become a burden for other projects 🙃

Reply:
The slf4j testing did turn in to a complete clownshow, yes. I was hoping that by this point in the year, as Java 15 came out and whatnot, we'd see slf4j 2.0 stabilize and then we'd have better options, but... turns out that hoping people you don't know will do the work you want doesn't necessarily get results.

Testing log output seemed important at the time because that was really the only way that code communicated its outputs.  That could be something to keep in mind for future iterations: "my slf4j logs in this TextArea" is currently the only user interface to some important information. But as Launcher develops a richer user interface to communicate about its status and troubleshoot failures, it'll also be developing APIs for that information that don't require going through the logger, and we can base tests on those instead.

Reply:
Back to the present, though: I hope _someone_ can confirm that TestRunGameTask is comprehensible, because this is where all the persnickity stuff about process state and inter-thread communication is. If things go wrong, or we discover other edge-cases on other platforms, this is where the work will be.

Reply:
I can confirm that it _looks_ comprehensible 😉 it's nicely structured, has comments, and so on. I think I should have put more emphasis on "in detail" rather than "not reviewing" somehow 🙃